src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -499,6 +499,53 @@ internal void Insert(ref T[] list, string? name, MemberListType listType)
                     }
                 }
 
+                private void MergeWithGlobalListInOrder(T[] list)
+                {
+                    T[]? cachedMembers = m_allMembers;
+
+                    if (cachedMembers == null)
+                    {
+                        m_allMembers = list;
+                        return;
+                    }
+
+                    int length = list.Length;
+                    T[] newCache = list;
+
+                    // When all members are requested there should be no element in cache that not found within all members list.
+                    // But somehow in very rare case such element found in cache and not appending that to the all members list
```

Comment:
What are these very rare cases? Can we fix the bug that this is referring to?

Reply:
I was not able to find clear repro, it causes random test failures in random projects, I thought this is known issue related to the comment below:
https://github.com/dotnet/runtime/blob/5891a41d36b1e05c0fed78e4648105cac8a81776/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L591-L602 
Me and @GrabYourPitchforks tried to find that ticket `DevDiv #339308` but could not. I will try to repro and see if I can find out what case is that

Reply:
I think these extra MethodInfo are most likely the MethodInfos created from MethodDescs that are not normalized for reflection: https://github.com/dotnet/runtime/issues/67989#issuecomment-1133246061. It may be a good idea to fix #67989 first, so that we do not need to worry about those non-normalized MethodInfos here - it is hard to tell what kind of effect they may have on the order.

Reply:
Right, completely agree